articles/libraries/machine-learning/design.md

```diff
@@ -0,0 +1,54 @@
+
+- A large number of parameters implies a more flexible model, which may be suitable to draw complicated classification boundaries but also more susceptible to overfitting.
+
+- Entangling gates between qubits are essential to capture the correlations between the quantum features.
+
+## How to build a classifier with Q\#
+
+To build a classifier we are going to concatenate parametrized controlled rotations in our circuit model. To do it we can use the type [`ControlledRotation`](link to API article) defined in the Quantum Machine Learning library. This type accepts four arguments that determine: the index of the target qubit, the array of indices of the control qubits, the axis of rotation, and index of the associated parameter in the array of parameters defining the model.
+
+Let's see an example of a classifier. In the [half-moons sample](link to the sample) we can find the following classifier defined in the file `Training.qs`.
+
+    ```qsharp
+    function ClassifierStructure() : ControlledRotation[] {
+        return [
+            ControlledRotation((0, new Int[0]), PauliX, 4),
+            ControlledRotation((0, new Int[0]), PauliZ, 5),
+            ControlledRotation((1, new Int[0]), PauliX, 6),
+            ControlledRotation((1, new Int[0]), PauliZ, 7),
+            ControlledRotation((0, [1]), PauliX, 0),
+            ControlledRotation((1, [0]), PauliX, 1),
+            ControlledRotation((1, new Int[0]), PauliZ, 2),
+            ControlledRotation((1, new Int[0]), PauliX, 3)
+        ];
+    }
+    ```
+
+What we are defining here is a function that returns an array of `ControlledRotation` elements, that together with an array of parameters and a bias will define our [`SequentialModel`](link to API). This type is fundamental in the Quantum Machine Learning library and defines the classifier. The circuit defined in the function above is part of a classifier in which each sample of the dataset contains two features. Therefore we only need two qubits. The graphical representation of the circuit is:
+
+ ![Circuit model example](~/media/circuit_model_1.PNG)
+
+## Use the library functions to write layers of gates
+
+Suppose we have a dataset with 784 features per instance, e.g. images of 28x28 pixels like the MNIST dataset. In this case, the width of the circuit becomes large enough so that writing by hand each individual gate becomes a possible but unpractical task. This is why the Quantum Machine Learning library provides a set of tools to automatically generate layers of parametrized rotations. For instance, the function [`LocalRotationsLayer`](link to API) returns an array of uncontrolled single-qubit rotations along a given axis, with one rotation for each qubit in the register, parametrized by different model parameters. For example, `LocalRotationsLayer(4,X)` returns the following set of gates:
```

Comment:
```suggestion
Suppose we have a dataset with 784 features per instance, e.g. images of 28×28 pixels like the MNIST dataset. In this case, the width of the circuit becomes large enough so that writing by hand each individual gate becomes a possible but impractical task. This is why the Quantum Machine Learning library provides a set of tools to automatically generate layers of parametrized rotations. For instance, the function [`LocalRotationsLayer`](link to API) returns an array of uncontrolled single-qubit rotations along a given axis, with one rotation for each qubit in the register, each parametrized by a different model parameter. For example, `LocalRotationsLayer(4, X)` returns the following set of gates:
```

As a side note, keeping source lines shorter can help make more informative diffs and GitHub suggestions, such that I'd recommend breaking this line up a bit in the source.